linux/xdg/src/main/java/com/github/weisj/darkmode/platform/linux/xdg/XdgThemeMonitorService.kt

```diff
@@ -0,0 +1,37 @@
+// based on https://gist.github.com/DevSrSouza/b013d1a8119f50615a493b36cf0b9b56
+
+package com.github.weisj.darkmode.platform.linux.xdg
+
+import com.github.weisj.darkmode.platform.NativePointer
+import com.github.weisj.darkmode.platform.ThemeMonitorService
+import org.freedesktop.dbus.interfaces.DBusSigHandler
+
+class XdgThemeMonitorService : ThemeMonitorService {
+    private val freedesktopConnection = FreedesktopConnection()
+    private val sigHandler = SigHandler()
+    override val isDarkThemeEnabled: Boolean = freedesktopConnection.theme == ThemeMode.DARK
+    override val isSupported: Boolean = freedesktopConnection.theme != ThemeMode.ERROR
```

Comment:
Try this. Calling out to system resources during class initialization makes debugging deadlocks rather difficult. This way you'll at least know which call is causing the lock (It might not be the creation of the connection but the `theme` call instead)
```suggestion
    private val freedesktopConnection by lazy { FreedesktopConnection() }
    private val sigHandler = SigHandler()
    override val isDarkThemeEnabled: Boolean
        get() = freedesktopConnection.theme == ThemeMode.DARK
    override val isSupported: Boolean
        get() = freedesktopConnection.theme != ThemeMode.ERROR
```

Reply:
I'm pretty sure it's the creation of the connection. When I comment everything out except the creation, the deadlock is still there 😅

Reply:
You can’t comment out `isDarkThemeEnabled` and `isSupported` though as they need to be implemented in order for the class to compile. Did you stub them or what exactly do you mean by commenting out?
Also it would be beneficial to add some unit tests, testing whether it works in isolation outside IntelliJ.

Reply:
Is there a way to run IDE and plugin without the sandbox?

Reply:
> You can’t comment out isDarkThemeEnabled and isSupported though as they need to be implemented in order for the class to compile. Did you stub them or what exactly do you mean by commenting out?
Also, it would be beneficial to add some unit tests, testing whether it works in isolation outside IntelliJ.

Didn't see that, sorry :sweat_smile: 

I just returned true and commented the rest out. I have unittests that I run regularly, but I didn't commit them as I don't know which theme is actually active